pkg/apis/tenancy/v1alpha1/helper/helper.go

```diff
@@ -40,3 +40,12 @@ func QualifiedObjectName(obj metav1.Object) string {
 	}
 	return fmt.Sprintf("%s|%s", logicalcluster.From(obj), obj.GetName())
 }
+
+// IsReservedName indicates that the provided name is reserved and cannot be
+// used in a user-supplied object.
+func IsReservedName(name string) bool {
+	if name == "root" || name == "system" {
+		return true
+	}
+	return false
```

Comment:
simplification suggestion: `return name == "root" || name == "system"`

Reply:
Also: reserved clusterworkspace names are `root` and `system`. Reserved clusterworkspacetype names are `system` and `any`. (see the 2 `.yaml-patch` files in `config/crds` for the respective crds).